packages/subgraph/schema.graphql

```diff
@@ -1,181 +1,460 @@
 # Types
 
-# Token
-
+#############################
+########## Tokens ###########
+#############################
+
+"""
+Interface representing a generic token.
+Tokens implementing this interface are expected to provide certain standard properties and functionalities.
+"""
 interface Token {
-  id: ID! # use address as id
+  " The address of the Token Smart Contract. "
+  id: ID!
+
+  " The name of the token, mirrored from the smart contract. "
   name: String
+
+  " The symbol of the token, mirrored from the smart contract. "
   symbol: String
 }
 
+"""
+An [ERC20 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-20), type of Token.
+"""
 type ERC20Contract implements Token @entity(immutable: true) {
-  id: ID! # use address as id
+  " The address of the [ERC20 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-20). "
+  id: ID!
+
+  " The name of the token, mirrored from the smart contract. "
   name: String
+
+  " The symbol of the token, mirrored from the smart contract. "
   symbol: String
+
+  " The number of decimals of the token, mirrored from the smart contract. "
   decimals: Int!
 }
 
-type ERC20WrapperContract implements Token @entity(inmutable: true) {
-  id: ID! # use address as id
+"""
+Wrapped [ERC20 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-20). Used for token voting.
+"""
+type ERC20WrapperContract implements Token @entity(immutable: true) {
+  " The address of the [ERC20 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-20). "
+  id: ID!
+
+  " The name of the wrapped token, mirrored from the smart contract. "
   name: String
+
+  " The symbol of the wrapped token, mirrored from the smart contract. "
   symbol: String
+
+  " The number of decimals of the wrapped token, mirrored from the smart contract. "
   decimals: Int!
+
+  " The underlying [ERC20 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-20) this wrapper represents. "
   underlyingToken: ERC20Contract!
 }
 
+"""
+[ERC721 Token Standard](https://eips.ethereum.org/EIPS/eip-721), Non-Fungible Token (NFT), type of Token.
+"""
 type ERC721Contract implements Token @entity(immutable: true) {
-  id: ID! # use address as id
+  " The address of the [ERC721 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-721). "
+  id: ID!
+
+  " The name of the [ERC721 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-721). "
   name: String
+
+  " The symbol of the [ERC721 Token Smart Contract](https://eips.ethereum.org/EIPS/eip-721). "
   symbol: String
```

Comment:
the name & symbol is not of ERC721 contract, it will be of the token
the contract name could be different from the token name